pulsar-functions/api-java/src/main/java/org/apache/pulsar/functions/api/Record.java

```diff
@@ -77,6 +77,15 @@
         return Optional.empty();
     }
 
+    /**
+     * Retrieves the partition number if any of the record.
+     *
+     * @return The partition number
+     */
+    default Optional<Integer> getPartitionNumber() {
```

Comment:
```suggestion
    default Optional<Integer> getPartitionId() {
```

Reply:
I think parittion id it better than partition number

Reply:
@sijie I agree but `default Optional<String> getPartitionId() {` already defined on line 76.

Reply:
How about `getPartitionIndex`?

Reply:
@sijie renamed